File: amazia-core/src/main/java/com/example/delivery/controller/ListShippingLeadTimeController.java
package com.example.delivery.controller;

import com.example.delivery.dto.ShippingLeadTimeResponse;
import com.example.delivery.service.ListShippingLeadTimeService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 都道府県別リードタイム一覧 API（GET /api/shipping-lead-times[?shippingMethodId=N]）。
 */
@RestController
@RequestMapping("/api")
public class ListShippingLeadTimeController {

    private final ListShippingLeadTimeService service;

    public ListShippingLeadTimeController(ListShippingLeadTimeService service) {
        this.service = service;
    }

    @GetMapping("/shipping-lead-times")
    public ResponseEntity<List<ShippingLeadTimeResponse>> list(
            @RequestParam(required = false) Long shippingMethodId) {
        return ResponseEntity.ok(service.list(shippingMethodId));
    }
}

File: amazia-console/app/Delivery/Controller/ListShippingLeadTimeController.php
<?php

namespace App\Delivery\Controller;

use App\Delivery\Service\ListShippingLeadTimeService;
use App\Http\Controllers\Controller;
use Illuminate\Http\Request;

/**
 * 都道府県別リードタイム一覧 Controller（GET /api/shipping-lead-times[?shippingMethodId=N]）。
 *
 * 設計書 §機能詳細：approver_roles（supervisor 以上）のみ閲覧可。
 */
class ListShippingLeadTimeController extends Controller
{
    private ListShippingLeadTimeService $service;

    public function __construct(ListShippingLeadTimeService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request)
    {
        $role = (string) $request->get('auth_role');
        if (!in_array($role, config('app.auth.approver_roles'), true)) {
            return response()->json(['message' => 'Forbidden'], 403);
        }

        $shippingMethodId = $request->query('shippingMethodId');
        $response = $this->service->list($shippingMethodId === null ? null : (int) $shippingMethodId);
        return response()->json($response->json(), $response->status());
    }
}
